mlir/include/Gradient/Utils/EinsumLinalgGeneric.h

```diff
@@ -30,9 +30,9 @@ namespace catalyst {
 /// `b` and the `result` correspondingly. For example, the `([0,1,2,3], [2,3], [0,1])` codes would
 /// be equivalent to the following `np.einsum` format string: `"abcd,cd->ab"`.
 mlir::Value einsumLinalgGeneric(mlir::OpBuilder &builder, mlir::Location loc,
-                                llvm::ArrayRef<size_t> axisCodesA,
-                                llvm::ArrayRef<size_t> axisCodesB,
-                                llvm::ArrayRef<size_t> axisCodesResult, mlir::Value a,
-                                mlir::Value b);
+                                llvm::ArrayRef<int64_t> axisCodesA,
```

Comment:
The change from unsigned to signed integers to represent dimension sizes is necessary to handle dynamic sizes because internally, `ShapedType::kDynamic` is defined as `INT_MIN` (which is a negative number). This is required because the `einsumLinalgGeneric` function is used in the custom quantum gradient with a dynamic dimension (the number of gate parameters).